state/src/item/validator_set.rs

```diff
@@ -0,0 +1,112 @@
+// Copyright 2020 Kodebox, Inc.
+// This file is part of CodeChain.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+use super::stake::{CurrentValidators, NextValidators, Validator};
+use crate::{StateResult, TopLevelState};
+use ckey::Ed25519Public as Public;
+use ctypes::{CompactValidatorEntry, CompactValidatorSet};
+use std::ops::Deref;
+
+// Validator information just enough for the host
+pub struct SimpleValidator(Validator);
+
+impl SimpleValidator {
+    pub fn weight(&self) -> u64 {
+        self.0.delegation()
+    }
+
+    pub fn pubkey(&self) -> &Public {
+        self.0.pubkey()
+    }
+}
+
+// TODO: implementation will be changed as we move NextValidators into a separate module
+pub struct NextValidatorSet(NextValidators);
+
+impl NextValidatorSet {
+    // Validator::close_block will return validator_set
+    // The host should update its next validators from the information
+    pub fn from_compact_validator_set(validators: CompactValidatorSet) -> Self {
+        let validators: Vec<_> = validators
+            .deref()
```

Comment:
You don't need to call `deref`. The dot operator conducts an implicit dereferencing.

Reply:
Thanks. I removed it.